prow/cmd/deck/main.go

```diff
@@ -1150,7 +1150,36 @@ func handleProwJob(prowJobClient prowv1.ProwJobInterface) http.HandlerFunc {
 	}
 }
 
-func handleRerun(prowJobClient prowv1.ProwJobInterface, createProwJob bool) http.HandlerFunc {
+// canTriggerJob determines whether the given user can trigger any job.
+func canTriggerJob(user string, cfg config.RerunAuthConfig) bool {
+	if cfg.AllowAnyone {
+		return true
+	}
+	if cfg.AuthorizedUsers == nil {
```

Comment:
nit: `AuthorizedUsers` isn't a pointer so this check isn't necessary. Iterating over a `nil` slice is a safe no-op. https://play.golang.org/p/WkXhthckUXj